### PR TITLE
Add status endpoints

### PR DESCRIPTION
**Pull request by THenry14**
## Describe your changes

Added gateway/rpc status endpoints

## Linked issues

https://github.com/software-mansion/starknet-jvm/issues/57

## Breaking changes

- [ ] This issue contains breaking changes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -39,6 +39,10 @@ local.properties
 
 # OS specific files
 .DS_Store
+venv
 
 # Exclude gradle wrapper jar
 !gradle/wrapper/gradle-wrapper.jar
+
+# Test artifacts
+lib/src/test/resources
```
Comment (cptartur) [inline .gitignore:48]: I don't think we should be excluding all resources from git. We keep contracts used for tests there etc.

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt [hunk 2]
```diff
@@ -45,3 +45,53 @@ data class TransactionFailureReason(
...
+    override val startingBlockNumber: Int,
+    @JsonNames("current_block_hash")
+    override val currentBlockHash: Felt,
+    @JsonNames("current_block_num")
+    override val currentBlockNumber: Int,
+    @JsonNames("highest_block_hash")
+    override val highestBlockHash: Felt,
+    @JsonNames("highest_block_num")
+    override val highestBlockNumber: Int,
+) : Syncing()
```
Comment (cptartur) [inline lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt:97]: Small nitpick, but can you add newlines between each value as it is done in `Transaction.kt`? I think it improves readability when there are annotations on parameters.

File: lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/JsonRpcProvider.kt [hunk 3]
```diff
@@ -226,6 +227,54 @@ class JsonRpcProvider(
         return buildRequest(JsonRpcMethod.DECLARE, params, DeclareResponse.serializer())
     }
 
+    override fun getBlockNumber(): Request<Int> {
+        val params = Json.encodeToJsonElement(JsonArray(listOf()))
```
Comment (cptartur) [inline lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/JsonRpcProvider.kt:231]:
```suggestion
        val params = Json.encodeToJsonElement(JsonArray(emptyList()))
```

File: .gitignore (hunk 1 above)
Comment (THenry14) [reply .gitignore:48]: Oh wow, I went over the top with this. There is a file that is being created as a result of running tests - `starknet_open_zeppelin_accounts.json` which I wanted to exclude from git. So I guess the question is, should I exclude it here explicitly OR we should maybe get rid of it after tests conclude?

Comment (cptartur) [reply .gitignore:48]: I think we should exclude only `lib/src/test/resources/account/` directory. Removing it after tests conclude can be done additionally, but I think that's for another PR.

**Review by cptartur — APPROVED** [review]

**Pull request merged**